config_parser_test.cc

```diff
@@ -1,11 +1,79 @@
 #include "gtest/gtest.h"
 #include "config_parser.h"
 
+#include <sstream>
+#include <string>
+
 TEST(NginxConfigParserTest, SimpleConfig) {
-  NginxConfigParser parser;
-  NginxConfig out_config;
+    NginxConfigParser parser;
+    NginxConfig out_config;
+
+    bool success = parser.Parse("example_config", &out_config);
+
+    EXPECT_TRUE(success);
+}
+
+TEST(NginxConfigParserTest, ToStringSimple) {
+    NginxConfigStatement statement;
+    statement.tokens_.push_back("foo");
+    statement.tokens_.push_back("bar");
+    EXPECT_EQ("foo bar;\n", statement.ToString(0)) << 
+        "foo bar wasn't serialized correctly";
+}
+
+class NginxStringConfigTest : public ::testing::Test {
+protected:
+    bool ParseString(const std::string& config_string) {
+        std::stringstream config_stream(config_string);
+        return parser_.Parse(&config_stream, &config_);
+    }
+
+    NginxConfigParser parser_;
+    NginxConfig config_;
+};
+
+TEST_F(NginxStringConfigTest, SimpleConfigStream) {
+
+    EXPECT_TRUE(ParseString("foo bar;")) << "Didn't parse successfully";
+    EXPECT_EQ(1, config_.statements_.size()) << "Didn't get one statement";
+    EXPECT_EQ("foo", config_.statements_.at(0)->tokens_.at(0));
+}
+
+TEST_F(NginxStringConfigTest, EmptyConfig) {
 
-  bool success = parser.Parse("example_config", &out_config);
+    EXPECT_FALSE(ParseString("")) << "Parsed empty config";
+}
+
+TEST_F(NginxStringConfigTest, BadConfig) {
+    EXPECT_FALSE(ParseString("foo bar")) 
+        << "Parsed config missing semicolon correctly";
+}
+
+TEST_F(NginxStringConfigTest, NestedConfig) {
```

Comment:
you could check if the NginxConfig has the correct number of  elements in statements_